### PR TITLE
Fix for edit services throw error due to proxy data is undefined in storeSettings

### PR DESCRIPTION
**Pull request by ghost**
#### Pre-flight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

- fix: edit services throw error due to proxy data is undefined in cache

#### Motivation and Context

-  #238 

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by ghost** [comment]
Team - 
Welcome all to test this pr, and once @Alphrag test and confirm the fix is ok I shall merge this PR.

**Comment by Alphrag** [comment]
This temporary fix seems to work, but the root of the problem would be the fact that `proxy.isEnabled` returns true when it hasn't been set up. 
Would this be linked to #750 ?

File: src/containers/settings/EditServiceScreen.tsx [hunk 1]
```diff
@@ -344,9 +344,17 @@ class EditServiceScreen extends Component<IProps> {
...
+          TODO - [TS DEBT] find out why sometimes proxy[service.id] gives undefined
+          Note in proxy service id exist as key but value is undefined rather that proxy empty object
+
+          Temp fix - or-ed {} (to stores.settings.proxy[service.id] ) to avoid undefined proxy in settingStore throw error
+          */
+          stores.settings.proxy[service.id] || {}
         : {};
 
+      // @ts-ignore
+      console.log(stores.settings.proxy, service);
```
Comment (Alphrag) [inline src/containers/settings/EditServiceScreen.tsx]: Possibly worth removing this in order to avoid overloading the logs ;-)

Comment (ghost) [reply src/containers/settings/EditServiceScreen.tsx]: removed it and amended

**Comment by ghost** [comment]
> This temporary fix seems to work, but the root of the problem would be the fact that `proxy.isEnabled` returns true when it hasn't been set up. Would this be linked to #750 ?

I think the issue is from file src/features/serviceProxy/index.ts (line 49) as its taking care of setting proxy, In case we perform sequential proxy variable update instead of parallel we may know some error.

My thought process is issue #750 handle default values which is shown in UI (it seems to be correct and in off state for newly created service)

**Review by Alphrag — APPROVED** [review]

**Pull request merged**